server/src/main/java/io/crate/expression/symbol/Literal.java

```diff
@@ -270,4 +274,16 @@ public static Literal<Float> of(Float value) {
     public static Literal<Period> newInterval(Period value) {
         return new Literal<>(DataTypes.INTERVAL, value);
     }
+
+    public static Literal<?> childOf(Literal<Map<String, Object>> objectLiteral, List<String> path) {
+        assert objectLiteral.valueType().id() == ObjectType.ID : "An object literal expected";
+        ObjectType objectType = (ObjectType) objectLiteral.valueType();
+        DataType<?> childType = objectType.innerType(path);
+        Object childValue = Maps.getByPath(objectLiteral.value(), path);
+        if (childType.id() == UNDEFINED.id()) {
+            // TODO: inner types are missing when the parent object type is returned from a function
+            childType = guessType(childValue);
```

Comment:
Eventually we should fix and I think via [#14828](https://github.com/crate/crate/issues/14828) planned for `6.0`.